Craigslist/Craigslist/Controllers/ListingController.cs

```diff
@@ -13,9 +14,11 @@ public class ListingController : Controller
 		private readonly LookupManager lookupManager = new LookupManager();
 		private readonly CategoriesHelper categoriesHelper = new CategoriesHelper();
 
-	    public ViewResult List(long? categoryId = null)
+        public ViewResult List(int? page, long? categoryId = null )
 	    {
-		    return View(listingsManager.GetListingsByCategoryId(categoryId));
+            int pageSize = 1;
+            int pageNumber = (page ?? 1);
+		    return View(listingsManager.GetListingsByCategoryId(categoryId).ToPagedList(pageNumber, pageSize));
```

Comment:
Do you understand what you are doing here?
Imaging you have 100000 items. You're fetching all of them into server memory and then applying pagination. 
i'll take a look at the implementation of ToPagedList method but i'm pretty sure that it just applies Skip and Take LINQ methods behind the scene. 
It's not acceptable. 
Skip and Take should be applied withing the opened domain context in order to get translated into SQL top() and row_number() instructions.